code/game/gamemodes/gang/gang.dm

```diff
@@ -0,0 +1,366 @@
+			if(!antag_candidates.len)
+				log_game("RAN OUT OF REPLACMENT GANGSTERS!!! SHIT GOING TO BE FUCKED!!!")
+				break
+			var/datum/mind/gangbanger = antag_pick(antag_candidates)
+			gangbangers += gangbanger
+			log_game("[key_name(gangbanger)] has been selected as a replacement gangster!")
+	for(var/datum/mind/gangbanger in gangbangers)
+		var/gang_to_use = pick_n_take(gangs_to_use)
+		var/datum/antagonist/gang/new_gangster = new gang_to_use()
+		var/datum/team/gang/ballas = new /datum/team/gang()
+		new_gangster.my_gang = ballas
+		gangs += ballas
+		ballas.add_member(gangbanger)
+		ballas.name = new_gangster.gang_name
+
+		ballas.acceptable_clothes = new_gangster.acceptable_clothes.Copy()
+		ballas.free_clothes = new_gangster.free_clothes.Copy()
+
+		for(var/C in ballas.free_clothes)
+			var/obj/O = new C(gangbanger.current)
+			var/list/slots = list (
+				"backpack" = ITEM_SLOT_BACKPACK,
+				"left pocket" = ITEM_SLOT_LPOCKET,
+				"right pocket" = ITEM_SLOT_RPOCKET
+			)
+			var/mob/living/carbon/human/H = gangbanger.current
+			var/equipped = H.equip_in_one_of_slots(O, slots)
+			if(!equipped)
+				to_chat(gangbanger.current, "Unfortunately, you could not bring your [O] to this shift. You will need to find one.")
+				qdel(O)
+
+		gangbanger.add_antag_datum(new_gangster)
+		gangbanger.current.playsound_local(get_turf(gangbanger.current), 'sound/ambience/antag/thatshowfamiliesworks.ogg', 100, FALSE, pressure_affected = FALSE)
+		to_chat(gangbanger.current, "<B>As you're the first gangster, your uniform and spraycan are in your inventory!</B>")
+	addtimer(CALLBACK(src, .proc/announce_gang_locations), 5 MINUTES)
+	addtimer(CALLBACK(src, .proc/five_minute_warning), time_to_end - 5 MINUTES)
+	gamemode_ready = TRUE
+	SSshuttle.registerHostileEnvironment(src)
+	..()
+
+/datum/game_mode/gang/proc/announce_gang_locations()
+	var/list/readable_gang_names = list()
+	for(var/datum/team/gang/G in gangs)
+		readable_gang_names += "[G.name]"
+	var/finalized_gang_names = english_list(readable_gang_names)
+	priority_announce("Julio G coming to you live from Radio Los Spess! We've been hearing reports of gang activity on [station_name()], with the [finalized_gang_names] duking it out, looking for fresh territory and drugs to sling! Stay safe out there for the hour 'till the space cops get there, and keep it cool, yeah? Play music, not gunshots, I say. Peace out!", "Radio Los Spess", 'sound/voice/beepsky/radio.ogg')
+	sent_announcement = TRUE
+
+/datum/game_mode/gang/proc/five_minute_warning()
+	priority_announce("Julio G coming to you live from Radio Los Spess! The space cops are closing in on [station_name()] and will arrive in about 5 minutes! Better clear on out of there if you don't want to get hurt!", "Radio Los Spess", 'sound/voice/beepsky/radio.ogg')
+
+/datum/game_mode/gang/check_win()
+	var/alive_gangsters = 0
+	var/alive_cops = 0
+	for(var/datum/mind/gangbanger in gangbangers)
+		if(gangbanger.current)
```

Comment:
No need for this check. ishuman already checks if it's null.

Reply:
huh, the more you know
👍